### PR TITLE
feat: update types

### PR DESCRIPTION
**Pull request by jablko**
This PR adds some missing fields and fixes or improves the types of some others:
- The `bugs` field [can have an `email` sub-field](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bugs). You can specify the value for `bugs` as a simple string instead of an object.
- The `bugs`, `author`, `contributors`, and `repository` fields can be simple strings in package.json, but not in registry metadata. What I did was:
  - Move the `string` alternative from the `Maintainer` and `Repository` types to the `PackageJson` `author`, `contributors`, and `repository` field declarations.
  - In the `PackumentVersion` sub-type, override the `bugs`, `author`, `contributors`, and `repository` fields to exclude the `string` alternative.
- The `Maintainer` `name` and `Repository` `url` sub-fields are required. `repository` [can have a `directory` sub-field](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#repository).
- The `bundledDependencies` field [is a list of strings](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bundleddependencies), not an object like the other dependecies fields. Same with its `bundleDependencies` spelling.
- Add the [`os`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#os), [`cpu`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#cpu), and [`users`](https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#:~:text=users%3A%20an%20object%20whose%20keys%20are%20the%20npm%20user%20names%20of%20people%20who%20have%20starred%20this%20package) fields.
- Every package [will have a `latest` tag defined](https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#:~:text=every%20package%20will%20have%20a%20latest%20tag%20defined.), so it's required.

## References

Depends on https://github.com/npm/types/pull/16. I based this on https://github.com/npm/types/pull/16 to avoid having to repeat the changes for the hoisted and abbreviated copies of each field.

### CODE REVIEW

**Comment by broofa** [comment]
@darcyclarke @chrisdickinson  Apologies for spamming everyone's inbox here, but it's unclear to me whether this repository is still maintained.  So... I'm being the "squeaky wheel" in the hopes we can do something about it.

The types this repo provides are really useful.  I would argue their indispensable for anyone actiely using the NPM registry API, as they're provide much better and more accurate information than the registery docs.  They've been tremendously helpful in my [npmgraph](https://github.com/npmgraph/npmgraph) project.  But they're also clearly a bit long in the tooth and in need of an update, as @jablko's PRs and my own humble one seem to indicate.

It's a bit disheartening to see that there's been no response, comment, or movement on any of @jabiko's work.  Forking this repo is obviously an option, but that's a pretty sub-optimal outcome given that this is an official NPM repository.

So... can we get some eyes on this work?  What can I do to help out?

**Review by broofa — APPROVED** [review]
@jablko : You might want to edit this and your other PR's to set the "base branch" to point at the preceding branch you mention in each PR, rather than `master` here.  That should restrict the changes shown to just what's in the current branch rather.

**Comment by chrisdickinson** [comment]
Hi! @darcyclarke will know better than me – I left npm in 2018, so I don't have access to update this project anymore 😅

**Comment by broofa** [comment]
Pinging @dhei @lukekarrys @wraithgar @feelepxyz on [this](https://github.com/npm/types/pull/18#issuecomment-1823705507), as you four are at the intersection of `npm owner`s and GH `npm` org membership.

tl;dr:  Who's responsible for maintaining and publishing this project?  And does `npm` need help with it?

**Comment by broofa** [comment]
Broadening the "ping" here to include other NPM org members:  @arisacoba
@dylanatsmith @GrantBirki @joseph-lozano @joshspicer @pwelch @steiza @strackoverflow @tiagonbotelho @vcsjones @xodene

See [my previous comment] (https://github.com/npm/types/pull/18#issuecomment-1823705507) for more context.

tl;dr  Is this repo maintained?  If so, who should we be looking to for action on PRs like this one.  If not, would it be possible to add me as maintainer-collaborator to help move this work forward?

**Comment by lukekarrys** [comment]
Thanks for the ping. I did not know this repo existed before this (or I did and completely forgot).

It's definitely not being maintained currently but I will bring this up to our team and see if bringing it back to maintenance is something we can do.
